src/components/DateInput.js

```diff
@@ -0,0 +1,299 @@
+                                            Weeks
+                                        </Typography>
+                                    </Grid>
+                                    <Grid item xs={2}>
+                                        <Button 
+                                            color='primary'
+                                            variant={`${selectedWeek == 1 
+                                                ? 'contained' 
+                                                : 'outlined'
+                                            }`}
+                                            onClick={() => getWeekMonthYears(1)}
+                                        >
+                                            1
+                                        </Button>
+                                    </Grid>
+                                    <Grid item xs={2}>
+                                        <Button 
+                                            color='primary'
+                                            variant={`${selectedWeek == 2 
+                                                ? 'contained' 
+                                                : 'outlined'
+                                            }`}
+                                            onClick={() => getWeekMonthYears(2)}
+                                        > 
+                                            2
+                                        </Button>
+                                    </Grid>
+                                    <Grid item xs={2}>
+                                        <Button 
+                                            color='primary'
+                                            variant={`${selectedWeek == 3 
+                                                ? 'contained' 
+                                                : 'outlined'
+                                            }`}
+                                            onClick={() => getWeekMonthYears(3)}
+                                        >
+                                            3
+                                        </Button>
+                                    </Grid>
+                                    <Grid item xs={2}>
+                                        <Button 
+                                            color='primary'
+                                            variant={`${selectedWeek == 4 
+                                                ? 'contained' 
+                                                : 'outlined'
+                                            }`}
+                                            onClick={() => getWeekMonthYears(4)}
+                                        >
+                                            4
+                                        </Button>
+                                    </Grid>
+                                    <Grid item xs={2} />
```

Comment:
To make a more maintainable code do the following
```
const renderWeeks = () => {
   const weeks = [1,2,3,4]
   return weeks.map(w => {
      <Grid item xs={2}>
            <Button 
                    color='primary'
                     variant={`${selectedWeek == w 
                          ? 'contained' 
                           : 'outlined'
                        }`}
                         onClick={() => getWeekMonthYears(w)}
             >
                           {w}
             </Button>
           </Grid>
        <Grid item xs={2} />

   }
}
```

Now all you have to do is call the function `renderWeeks`, this way you are saving a lot of lines of code and if we want to change the weeks for any reason we only have to change the array. Do the same for months and years.

Reply:
Thanks Sofia for this method, the code now looks cleaner!